code/modules/mob/living/silicon/robot/robot_items.dm

```diff
@@ -1,3 +1,533 @@
+	set desc = "Allows you to set the Bodybag Containment Unit to deploy cryobags or bodybags."
+	set category = "Robot Commands"
+
+	var modeinput = input("Which type of bag should the BCU deploy?") in list("stasis bag","bodybag")
+	if(modeinput)
+		mode = modeinput
+		playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
+		if(mode == "stasis bag")
+			icon_state = "bodybag_store_cryo"
+		else if(mode == "bodybag")
+			icon_state = "bodybag_closed"
+		src.loc << "<span class='notice'>[src] set to [mode] deployment mode.</span>"
+
+/obj/item/weapon/cyborg/bodybag_containment_unit/examine(mob/user as mob)
+	if(..(src.loc == user))
+		user << "The [src] currently holds [bodybags_stored] bodybags and [cryobags_stored] cryobags. It is currently set to deploy [mode]s."
+
+/obj/item/weapon/reagent_containers/robot/chemvac
+	name = "ChemVac"
+	desc = "An industrial grade chemical distribution and reallocation device."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "chemvac"
+	amount_per_transfer_from_this = 50
+	possible_transfer_amounts = list(5,10,20,25,50,100)
+	volume = 100
+	var/mode = 1 //Initalized to draw reagents
+	update_icon()
+
+	attack_self(mob/user as mob)
+
+		switch(mode)
+			if(0)//Dispense Reagents
+				playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
+				user << "\blue Your ChemVac has been set to draw reagents."
+				mode = 1
+			if(1)//Draw Reagents
+				playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
+				user << "\blue Your ChemVac has been set to dispense reagents."
+				mode = 0
+		update_icon()
+
+	afterattack(obj/target, mob/user , flag)
+		switch(mode)
+			if(0)//Draw Reagents
+
+				if(!target.reagents) return
+
+				if(target.reagents.total_volume >= target.reagents.maximum_volume)
+					user << "\red [target] is full."
+					return
+
+				if(!target.is_open_container() && !ismob(target) && !istype(target,/obj/item/weapon/reagent_containers/food) && !istype(target, /obj/item/clothing/mask/cigarette)) //You can inject humans and food but you cant remove the shit.
+					user << "\red You cannot directly fill this object."
+					return
+
+				var/trans = 0
+
+				if(ismob(target))
+					if(istype(target , /mob/living/carbon/human))
+						var/mob/living/carbon/human/victim = target
+
+						var/obj/item/safe_thing = null
+						if( victim.wear_mask )
+							if ( victim.wear_mask.flags & MASKCOVERSEYES )
+								safe_thing = victim.wear_mask
+						if( victim.head )
+							if ( victim.head.flags & MASKCOVERSEYES )
+								safe_thing = victim.head
+						if(victim.glasses)
+							if ( !safe_thing )
+								safe_thing = victim.glasses
+
+						if(safe_thing)
+							if(!safe_thing.reagents)
+								safe_thing.create_reagents(100)
+							trans = src.reagents.trans_to(safe_thing, amount_per_transfer_from_this)
+
+							for(var/mob/O in viewers(world.view, user))
+								O.show_message(text("\red <B>[] tries to squirt something into []'s eyes, but fails!</B>", user, target), 1)
```

Comment:
What? What even is this? Using text is insane, you should just have [user] tries to squirt something into [target]'s eyes. And color macros!